proposals/readonly-setter-calls-on-non-variables.md

```diff
@@ -0,0 +1,127 @@
+# Readonly setter calls on non-variables
+
+Champion issue: <https://github.com/dotnet/csharplang/issues/9174>
+
+## Summary
+
+Permits a readonly setter to be called on _all_ non-variable expressions:
+
+```cs
+var c = new C();
+// Remove the current CS1612 error, because ArraySegment<T>.this is readonly:
+c.ArraySegmentProp[10] = new object();
+
+// The same restriction has *already* been lifted for invocation expressions:
+c.ArraySegmentMethod()[10] = new object();
+
+class C
+{
+    public ArraySegment<object> ArraySegmentProp { get; set; }
+    public ArraySegment<object> ArraySegmentMethod() => ArraySegmentProp;
+}
+```
+
+Currently, the code above gives the error CS1612 "Cannot modify the return value of 'C.ArraySegmentProp' because it is not a variable." This restriction is meaningless when the setter is readonly. The restriction is there to remind you to assign the modified struct value back to the property. But there is no modification to the struct value when the setter is readonly, so there is no reason to assign back to the property.
```

Comment:
```suggestion
Currently, the code above gives the error CS1612 "Cannot modify the return value of 'C.ArraySegmentProp' because it is not a variable." This restriction is unnecessary when the setter is readonly. The restriction is there to remind you to assign the modified struct value back to the property. But there is no supported modification to the struct value when the setter is readonly, so there is no reason to assign back to the property.
```

Suggest softening the language a bit. There are scenarios where mutations could occur, like `unsafe`, but they're not really in the realm of supported for these type of considerations.